app/lib/submission_builder/document.rb

```diff
@@ -56,28 +56,28 @@ def self.build(*args)
 
     def build_xml_doc(tag_name, **root_node_attributes)
       default_attributes = { 'xmlns:efile' => 'http://www.irs.gov/efile' }
-      return_state_attributes = { 'xmlns' => 'http://www.irs.gov/efile' }
+      return_state_attributes = get_return_state_attributes(tag_name)
       xml_builder = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
-        default_attributes.merge!(return_state_attributes) if merge_state_attrs_for_tag?(tag_name)
+        default_attributes.merge!(return_state_attributes) if return_state_attributes
         xml.send(tag_name, default_attributes.merge(root_node_attributes)) do |contents_builder|
           yield contents_builder if block_given?
         end
       end
       xml_builder.doc
     end
 
-    # This method is requirement from NY to remove the namespace attribute from
-    # any tag unrelated to the root XML tag. StateSubmissionManifest seems to
-    # need it also in order for the XML to be valid.
-    def merge_state_attrs_for_tag?(tag_name)
-      if self.submission.data_source_type == 'StateFileNyIntake'
-        if %w[ReturnState efile:ReturnState StateSubmissionManifest].include?(tag_name)
-          return true
-        else
-          return false
-        end
+    def get_return_state_attributes(tag_name)
+      return_state_attributes = { 'xmlns' => 'http://www.irs.gov/efile' }
+      intake_class = self.submission.data_source_type
+      if intake_class == 'StateFileNyIntake' && !%w[ReturnState efile:ReturnState StateSubmissionManifest].include?(tag_name)
+        # This method is requirement from NY to remove the namespace attribute from
+        # any tag unrelated to the root XML tag. StateSubmissionManifest seems to
+        # need it also in order for the XML to be valid.
+        return false
+      elsif intake_class == 'StateFileNjIntake' && (Rails.env.production? || Rails.env.demo?)
```

Comment:
[dust] do we want to remove the `&& (Rails.env.production? || Rails.env.demo?)`, either now or eventually?

Reply:
We need to leave the default xml namespace on local and test so we can validate the xml